nx/lib/nx/lin_alg.ex

```diff
@@ -1117,60 +1117,79 @@ defmodule Nx.LinAlg do
         48.0
       >
   """
-  def determinant(tensor)
+  defn determinant(tensor) do
+    Nx.Defn.Kernel.assert_shape_pattern(tensor, {n, n})
+
+    {n, _} = Nx.shape(tensor)
+
+    transform(n, fn
+      2 ->
+        determinant_2by2(tensor)
+
+      3 ->
+        determinant_3by3(tensor)
+
+      _ ->
+        determinant_NbyN(tensor)
+    end)
```

Comment:
I've rewritten all of these defnps to a more "defn-like" syntax, instead of "elixir-like". Let me know if I should roll this back